engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java

```diff
@@ -212,4 +213,19 @@ public static String collectExceptionMessages(Throwable cause) {
 
     return message;
   }
+
+  public static <T> T throwPersistenceException(Supplier<T> supplier) {
```

Comment:
The name of the method suggests that it always throws an exception. Can we rename it to something like `doWithExceptionWrapper`?

Also, let's add a Javadoc comment here why we are wrapping these exceptions. This will help other developers understand and refactor this in the future.

Reply:
👍